test_tableslurp.sh

```diff
@@ -0,0 +1,142 @@
+# /tmp/tablesnap-test/index.txt
+TEST_FILE="test_aws_s3_cp.txt"
+INDEX_FILE="index.txt"
+RENAMED_FILE="test_file_orig.txt"
+CONTAINER="tester"
+SRC_LOCAL="${DIR_LOCAL}/${TEST_FILE}"
+INDEX_SRC="${DIR_LOCAL}/${INDEX_FILE}"
+INDEX_BUCKET="${BUCKET}"
+BACKUP_BUCKET="${BUCKET}"
+
+# Create destinations on S3. Note: Dest for index file is at top-level,
+# i.e., not under a directory.
+# s3://kimlew-tablesnap-tests/tester:/tmp/tablesnap-test/test_aws_s3_cp.txt
+DEST_ON_S3=s3://"${BUCKET}"/"${CONTAINER}":"${DIR_LOCAL}"/"${TEST_FILE}"
+INDEX_DEST=s3://"${BUCKET}"/"${CONTAINER}":/"${INDEX_FILE}"
+
+echo "SRC_LOCAL is:" "${SRC_LOCAL}"
+echo "DEST_ON_S3 is:" "${DEST_ON_S3}"
+echo "INDEX_SRC is:" "${INDEX_SRC}"
+echo "INDEX_DEST is:" "${INDEX_DEST}"
+echo
+
+touch "${SRC_LOCAL}" > /dev/null
+touch "${INDEX_SRC}" > /dev/null
+
+echo "${DIR_LOCAL}/${TEST_FILE}" > "${INDEX_SRC}"
+echo "AFTER touch & > index file, the ls of /tmp/tablesnap-test:"
+ls -lah "${DIR_LOCAL}"
+echo "Contents of INDEX_FILE: "
+cat "${INDEX_SRC}"
+echo
+
+aws s3 cp "${SRC_LOCAL}" "${DEST_ON_S3}"
+aws s3 cp "${INDEX_SRC}" "${INDEX_DEST}"
+
+# Check if test file is IN /tmp/tablesnap-test.
+if aws s3 ls "${BUCKET}/${CONTAINER}:${DIR_LOCAL}"/ | grep "$TEST_FILE"; then
+	echo "SUCCESS: TEST_FILE has been copied to S3."
+else
+	echo "FAILURE: TEST_FILE has NOT been copied to S3."
+fi
+
+# Check if index file is IN /tmp/tablesnap-test.
+if aws s3 ls "${BUCKET}/${INDEX_FILE}" | grep "$INDEX_FILE"; then
+	echo "SUCCESS: INDEX_FILE has been copied to S3."
+else
+	echo "FAILURE: INDEX_FILE has NOT been copied to S3."
+fi
+echo
+
+# Get/Retrieve/Download file(s) from S3 bucket on AWS from this directory:
+# s3://kimlew-tablesnap-tests/tester:/tmp/tablesnap-test/test_aws_s3_cp.txt
+
+# Run tableslurp with current & extra 4 args. Then do a check.
+# --name tester - is the flag & the replacement container ID, "${CONTAINER}"
+# index_file is: index.txt
+# index_bucket is: kimlew-tablesnap-tests
+# backup_bucket is: kimlew-tablesnap-tests
+./${PYTHON_ENV}/bin/python ./tableslurp -k "$AWS_ACC_KEY_ID" -s "$AWS_SEC_ACC_KEY" --name "${CONTAINER}" "${INDEX_BUCKET}" "${INDEX_FILE}" "${BACKUP_BUCKET}"
+echo
+echo "Running tableslurp."
+
+# Rename the local file with mv to same directory. Check removal. Then re-run
+# tableslurp & check if file exists to prove file was retrieved from AWS.
+echo "${DIR_LOCAL}/${TEST_FILE}"
+echo "${DIR_LOCAL}/$RENAMED_FILE"
+mv "${DIR_LOCAL}/${TEST_FILE}" "${DIR_LOCAL}/${RENAMED_FILE}"
+
+# Check if test file is temporarily removed from /tmp/tablesnap-test.
+if [ -f "${DIR_LOCAL}/${TEST_FILE}" ]; then
+  echo "${TEST_FILE} was NOT deleted. Directory has:"
+  ls "${DIR_LOCAL}"
+else
+  echo "${TEST_FILE} has been deleted. Directory now has:"
+  ls "${DIR_LOCAL}"
+fi
+echo
+
+./${PYTHON_ENV}/bin/python ./tableslurp -k "$AWS_ACC_KEY_ID" -s "$AWS_SEC_ACC_KEY" --name "${CONTAINER}" "${INDEX_BUCKET}" "${INDEX_FILE}" "${BACKUP_BUCKET}"
+echo
+echo "Re-running tableslurp."
```

Comment:
This should be the only time you need to run `tableslurp`

Reply:
Right. As I mentioned in a few comments above, the first tableslurp run is deleted.

Reply:
Confirmed you're running `tableslurp` just once now.